politeiad/backendv2/tstorebe/tstorebe.go

```diff
@@ -1017,21 +1018,79 @@ func (t *tstoreBackend) Fsck() error {
 		return err
 	}
 
-	// Rebuild the inventory cache. This is a multi-step process:
-	//
-	// 1. Sort the tokens into two groups; unvetted and vetted. Each
-	//    group is additionally sorted by the timestamp of their most
-	//    recent status change.
-	//
-	// 2. Add the vetted tokens to the inventory cache. They MUST be
-	//    added starting with the oldest timestamp. The vetted token
-	//    is first added to the unvetted cache then moved to the vetted
-	//    cache. This is a limitation of the inventory API and mimicks
-	//    how records are added and updated when using the politeiad
-	//    API.
-	//
-	// 3. Add the unvetted tokens to the inventory cache. They MUST be
-	//    added starting with the oldest timestamp.
+	// Get the partial record for all tokens. This also guarantees that all
+	// tokens being manipulated actually correspond to a record on the backend.
+	records := make(map[string]*backend.Record, len(allTokens))
+	for _, token := range allTokens {
+		r, err := t.tstore.RecordPartial(token, 0, nil, true)
+		if err != nil {
+			return err
+		}
+		records[r.RecordMetadata.Token] = r
+	}
+
+	// Sort records into vetted and unvetted groups.
+	vetted := make([]*backend.Record, 0, len(allTokens))
```

Comment:
Prefer using `var ()` so that go fmt aligns it nicely.

```
	var (
		vetted   = make([]*backend.Record, 0, len(allTokens))
		unvetted = make([]*backend.Record, 0, len(allTokens))
	)
```

Reply:
much better